packages/workflow/src/Workflow.ts

```diff
@@ -1327,12 +1329,31 @@ export class Workflow {
 				additionalData,
 				executionData,
 				mode,
+				closeFunctions,
 				abortSignal,
 			);
 			const data =
 				nodeType instanceof Node
 					? await nodeType.execute(context)
 					: await nodeType.execute.call(context);
+
+			const closingErrors = [];
+			for (const closeFunction of closeFunctions) {
+				try {
+					await closeFunction();
+				} catch (error) {
+					closingErrors.push(error);
+				}
+			}
+
+			if (closingErrors.length > 0) {
+				throw new ApplicationError("Error on execution node's close function(s)", {
+					extra: { nodeName: node.name },
+					tags: { nodeType: node.type },
+					cause: closingErrors,
+				});
+			}
```

Comment:
Suggestion: We could use `Promise.allSettled` here and check whether any of them failed, maybe even aggregating the errors if necessary. Not a blocker tho.

Reply:
Good points, adjusted